geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/GeometryCalculations.java

```diff
@@ -0,0 +1,181 @@
+package mil.nga.giat.geowave.analytics.tools;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import javax.measure.quantity.Length;
+import javax.measure.unit.SI;
+import javax.measure.unit.Unit;
+
+import org.geotools.geometry.jts.ReferencedEnvelope;
+import org.geotools.referencing.GeodeticCalculator;
+import org.opengis.geometry.DirectPosition;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.GeometryFactory;
+
+public class GeometryCalculations
+{
+
+	final GeometryFactory factory;
+	final CoordinateReferenceSystem crs;
+	final double xMin, yMin, xMax, yMax;
+
+	public GeometryCalculations(
+			CoordinateReferenceSystem crs ) {
+		factory = new GeometryFactory();
+		this.crs = crs;
+		xMin = crs.getCoordinateSystem().getAxis(
+				0).getMinimumValue();
+		xMax = crs.getCoordinateSystem().getAxis(
+				0).getMaximumValue();
+		yMin = crs.getCoordinateSystem().getAxis(
+				1).getMinimumValue();
+		yMax = crs.getCoordinateSystem().getAxis(
+				1).getMaximumValue();
+	}
+
+	/**
+	 * Build geometries with the provided coordinate at the center. The width of
+	 * the geometry is twice the distance provided. More than one geometry is
+	 * return when passing the date line.
+	 * 
+	 * @param distances
+	 *            [x,y] = [longitude, latitude]
+	 * @param unit
+	 * @param coordinate
+	 * @return
+	 */
+	public List<Geometry> buildSurroundingGeometries(
+			final double[] distances,
+			final Unit<Length> unit,
+			Coordinate coordinate ) {
+		List<Geometry> geos = new LinkedList<Geometry>();
+		GeodeticCalculator geoCalc = new GeodeticCalculator();
```

Comment:
I move we don't use the geotools geodetic calculator;  if you do you need to guard / give up with antipodal pairs.  See: https://github.com/chrisbennight/geodesy - I think GeographicLib has a pure java implementation that's better on pretty much all counts.   It's not in maven central, but is MIT licensed